pkg/controller/route/labels.go

```diff
@@ -103,13 +115,34 @@ func (c *Controller) deleteLabelForOutsideOfGivenConfigurations(
 	// Delete label for newly removed configurations as traffic target.
 	for _, config := range oldConfigsList {
 		if _, ok := configMap[config.Name]; !ok {
+
 			delete(config.Labels, serving.RouteLabelKey)
-			if _, err := c.ServingClientSet.ServingV1alpha1().Configurations(ns).Update(config); err != nil {
-				logger.Errorf("Failed to update Configuration %s: %s", config.Name, err)
+
+			configClient := c.ServingClientSet.ServingV1alpha1().Configurations(config.Namespace)
+			if err := setRouteLabelForConfiguration(configClient, config.Name, nil); err != nil {
+				logger.Errorf("Failed to remove route label to configuration %q: %s", config.Name, err)
 				return err
 			}
 		}
 	}
 
 	return nil
 }
+
+func setRouteLabelForConfiguration(configClient servingv1alpha1.ConfigurationInterface, configName string, routeName *string) error {
```

Comment:
In principle, we should be able to generalize this into a helper in `pkg/controller` as something that takes:
```go
type Patcher interface {
    Patch(name string, pt types.PatchType(?), patch map[string]interface{})
}
```

You might also want this function to take the `map[string]interface{}` of `"labels"`...

REVISIT: As I realized I omitted the return type, I realized that the signature variance would likely be our downfall. :(

Reply:
> I realized that the signature variance would likely be our downfall. :(

der untergang